src/main/kotlin/pt/up/fe/ni/website/backend/controller/PostController.kt

```diff
@@ -18,18 +18,21 @@ class PostController(private val service: PostService) {
     @GetMapping
     fun getAllPosts() = service.getAllPosts()
 
-    @GetMapping("/{postId}")
+    @GetMapping("/{postId:[0-9]+}")
     fun getPost(@PathVariable postId: Long) = service.getPostById(postId)
 
     @PostMapping("/new")
     fun createPost(@RequestBody dto: PostDto) = service.createPost(dto)
 
-    @PutMapping("/{postId}")
+    @PutMapping("/{postId:[0-9]+}")
     fun updatePost(
         @PathVariable postId: Long,
         @RequestBody dto: PostDto
     ) = service.updatePostById(postId, dto)
 
-    @DeleteMapping("/{postId}")
+    @DeleteMapping("/{postId:[0-9]+}")
```

Comment:
For the same reason as below, I wouldn't include the regex if we don't have ambiguity. If the user introduces an invalid ID, it's less confusing to see "no post with that id" than "invalid endpoint". The users here are always our frontend friends :relaxed:

Reply:
For this, I kept the regex and if the parameter passed to the mapping isn't just numbers it'll be considered a slug and the error will be "post not found with slug X".

Reply:
That works well for the GET but not for PUT or DELETE, since slugs are not used in those cases